polaris-react/src/components/DataTable/DataTable.scss

```diff
@@ -201,6 +290,13 @@
   padding: var(--p-space-2);
   margin: var(--p-space-2);
 
+  #{$se23} & {
+    // stylelint-disable polaris/space/declaration-property-unit-disallowed-list -- Using hard coded px values to avoid cutting off the focus ring when overflow:hidden is applied via .Cell-truncate
+    margin: 3px;
+    padding: 9px 3px;
+    // stylelint-enable polaris/space/declaration-property-unit-disallowed-list
```

Comment:
I think we should establish a pattern to avoid hard coding these magic numbers (such as: custom properties for the focus ring width and offset). However, the px values are "needed" today to avoid cutting off the focus ring, maintain the existing row/column spacing, and allowing the remaining space for the tap target.

Before:
![Screenshot 2023-06-15 at 9 40 13 AM](https://github.com/Shopify/polaris/assets/32409546/e5a9f8d2-6e0f-4ff9-bf66-d8bab88d34b8)

After:
![Screenshot 2023-06-15 at 10 37 49 AM](https://github.com/Shopify/polaris/assets/32409546/47434e7c-31b7-41fa-8c72-6dbeffb22e3a)

After:

https://github.com/Shopify/polaris/assets/32409546/24866c71-3101-4f09-bcd6-02a64196303a